File: src/main/java/ar/edu/um/tesoreria/rest/model/ProveedorValor.java
/**
 * 
 */
package ar.edu.um.tesoreria.rest.model;

import java.io.Serializable;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

/**
 * @author daniel
 *
 */
@Data
@Entity
@Table(name = "movprov_detallevalores", uniqueConstraints = {
		@UniqueConstraint(columnNames = { "opv_orp_id", "opv_orden" }) })
@EqualsAndHashCode(callSuper = false)
@NoArgsConstructor
@AllArgsConstructor
public class ProveedorValor extends Auditable implements Serializable {
	/**
	 * 
	 */
	private static final long serialVersionUID = 2325757459322165022L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "opv_id")
	private Long proveedorValorId;

	@Column(name = "opv_orp_id")
	private Long proveedorMovimientoId;

	@Column(name = "opv_orden")
	private Integer orden;

	@Column(name = "opv_val_id")
	private Long valorMovimientoId;

}

File: src/main/java/ar/edu/um/tesoreria/rest/kotlin/model/ProveedorValor.kt
package ar.edu.um.tesoreria.rest.kotlin.model

import ar.edu.um.tesoreria.rest.model.Auditable
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import jakarta.persistence.UniqueConstraint


@Entity
@Table(
    name = "movprov_detallevalores",
    uniqueConstraints = [UniqueConstraint(columnNames = ["opv_orp_id", "opv_orden"])]
)
data class ProveedorValor(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "opv_id")
    var proveedorValorId: Long? = null,

    @Column(name = "opv_orp_id")
    var proveedorMovimientoId: Long? = null,

    @Column(name = "opv_orden")
    var orden: Int? = null,

    @Column(name = "opv_val_id")
    var valorMovimientoId: Long? = null,

    ) : Auditable()
